parts/k8s/kubernetesagentresourcesvmas.t

```diff
@@ -146,10 +146,14 @@
     },
 {{end}}
   {
+    {{if UserAssignedIDEnabled}}
+     "apiVersion": "[variables('apiVersionUserMSI')]",
```

Comment:
Does the API version above work w/ managed disks? The way this is implemented it assumes that the user MSI API version takes precedence over the managed disks API version.

Reply:
Testing was done with this and the managed disk set as below and it created the cluster fine. 

 "agentPoolProfiles": [
      {
        "name": "agentpool1",
        "count": 3,
        "vmSize": "Standard_D2_v2",
        "availabilityProfile": "VirtualMachineScaleSets",
        "storageProfile": "ManagedDisks"
      }

Is there a way to use one api version only for  parts of the specification - like for example to specify the identity in this case ?